src/cli_config.rs

```diff
@@ -31,6 +34,32 @@ impl FromStr for ThreadsConfig {
     }
 }
 
+impl<'de> Deserialize<'de> for ThreadsConfig {
```

Comment:
We don't write custom `Deserialize`, we should use only such a format that allows you put `derive` and use it.